src/App.tsx

```diff
@@ -15,36 +17,99 @@
   'Garlic',
 ];
 
+enum SortType {
+  String = 'sortByAlphabet',
+  Number = 'sortByLength',
+  Default = '',
+}
+
 export const App: React.FC = () => {
+  const [sortOrder, setSortOrder] = useState<SortType>(SortType.Default);
+  const [reversed, setReversed] = useState<boolean>(false);
+  const sortedGoods = [...goodsFromServer].sort(
+    (good1: string, good2: string) => {
+      switch (sortOrder) {
+        case 'sortByAlphabet':
+          return good1.localeCompare(good2);
+        case 'sortByLength':
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    },
+  );
+
+  const handleSortByAlphabet = useCallback(() => {
+    setSortOrder(SortType.String);
+  }, []);
+
+  const handleSortByLength = useCallback(() => {
+    setSortOrder(SortType.Number);
+  }, []);
+
+  const handleReverse = useCallback(() => {
+    setReversed(!reversed);
+  }, [reversed]);
+
+  const handleReset = useCallback(() => {
+    setSortOrder(SortType.Default);
+    setReversed(false);
+  }, []);
+
+  if (reversed) {
+    sortedGoods.reverse();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortOrder !== 'sortByAlphabet',
+          })}
+          onClick={handleSortByAlphabet}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortOrder !== 'sortByLength',
+          })}
+          onClick={handleSortByLength}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !reversed,
+          })}
+          onClick={handleReverse}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(reversed || sortOrder) && (
+          <button
+            type="button"
+            className="button is-danger"
+            onClick={handleReset}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {sortedGoods.map((good: string, id: number) => (
+          <li data-cy="Good" key={good + id}>
```

Comment:
[CODE KNOWLEDGE] - Good job on adding a unique `key` prop when rendering a list. This is important for React to correctly identify each item in the list.